Makefile

```diff
@@ -119,6 +119,11 @@ else
 	$(SED_I) "s/CROSS_BUILD_//g" $(DOCKERFILE)
 endif
 
+	# Update nginx template file
+ifeq ($(ARCH),arm)
+	$(SED_I) "s|x86_64-linux-gnu|arm-linux-gnueabihf|g" $(TEMP_DIR)/rootfs/etc/nginx/template/nginx.tmpl 
```

Comment:
@aledbf , a template is provided into the nginx-ingress-controller image when `COPY .` is executed from https://github.com/kubernetes/ingress-nginx/blob/master/rootfs/Dockerfile which is built as part in this Makefile per line 83 (`DOCKERFILE := $(TEMP_DIR)/rootfs/Dockerfile`).

Reply:
@toolboc that's not what @aledbf meant (he wrote that code btw), he was talking about the *base* nginx image. Just a little confusion I guess.
That said, `nginx.tmpl` does indeed contain hardcoded paths to `*.so` files. Maybe creating symlinks after the installation of Lua would be more elegant than replacing paths at build time using sed?

Reply:
@antoineco , we could fix this with symlinks via something like:
`ln -s  /usr/lib/arm-linux-gnueabihf /usr/lib/x86_64-linux-gnu` 

or 

we could symlink both of these paths to a new commonpath and adjust the template accordingly.  

That said, I feel like this is just getting around the issue being caused by the template itself.  Open to any suggestions and happy to follow up with an appropriate update.   

/Please advise

Reply:
> we could symlink both of these paths to a new commonpath and adjust the template accordingly

I like that one. Each container image is built for a specific architecture so using a specific path is not necessary.

Setting `LUA_PATH` and `LUA_CPATH` in the base nginx image is also something that would allow us to to skip the `lua_package_path` and `lua_package_cpath` directives (cf. [docs](https://github.com/openresty/lua-nginx-module#lua_package_path)).

Reply:
@antoineco , I have added a change which symlinks the appropriate paths for all platforms to a common `lua-platform-path` in  https://github.com/kubernetes/ingress-nginx/pull/2597/commits/ede1094f6d212b5f58d409eeed4d3c9b5747afdb

Reply:
@antoineco , It looks like the [CI build is getting stuck on changeset ede1094](https://travis-ci.org/kubernetes/ingress-nginx/builds/387912280) as it requires a republish of the base nginx image to account for the changes to build.sh which are now coupled to the modified nginx.tmpl.  Making the fix in the Makefile would bypass this issue.  This PR will not be able to move forward with the symlink strategy without an updated nginx image, bit of a chicken / egg problem.

Reply:
> Making the fix in the Makefile would bypass this issue. This PR will not be able to move forward with the symlink strategy without an updated nginx image, bit of a chicken / egg problem.

That is why we always make this change in two steps (PRs):
- change something nginx image (after the PR is merged I trigger the build for new image)
- change the main Makefile

Reply:
@aledbf ,

Thank you for the explanation, I will make a new PR for the changes to the nginx image, then we can make the change to nginx.tmpl here after the new image is published.

Let me know if there are any issues with this approach.